### PR TITLE
[FEATURE REQUEST] Create new file via Open in Web

### PR DESCRIPTION
**Pull request by JuancaG05**
## Related Issues
Implements https://github.com/owncloud/android/issues/3995

App PR: https://github.com/owncloud/android/pull/4023

### CODE REVIEW

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/CreateRemoteFileWithAppProviderOperation.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @JsonClass(generateAdapter = true)
+    data class CreateFileWithAppProviderResponse(
+        @Json(name = "file_id")
+        val fileId: String,
+    )
+
+    companion object {
+        private const val TIMEOUT: Long = 5_000
```
Comment (abelgardep) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/CreateRemoteFileWithAppProviderOperation.kt:106]: Would be great to unify timeouts across similar operations.

What do you think about that? Do you think it is feasible? 🤔

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/CreateRemoteFileWithAppProviderOperation.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import okhttp3.FormBody
+import okhttp3.RequestBody
+import timber.log.Timber
+import java.net.URL
+import java.util.concurrent.TimeUnit
+
+class CreateRemoteFileWithAppProviderOperation(
+    private val createFileWithAppProviderEndpoint: String,
+    private val parentContainerId: String,
+    private val filename: String,
```
Comment (abelgardep) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/CreateRemoteFileWithAppProviderOperation.kt:47]: Should we use camelCase in `fileName`?
```suggestion
    private val fileName: String,
```

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/GetUrlToOpenInWebRemoteOperation.kt [hunk 3]
```diff
@@ -76,7 +77,7 @@ class GetUrlToOpenInWebRemoteOperation(
         }
     }
 
-    private fun isSuccess(status: Int) = status == HttpConstants.HTTP_OK || status == HttpConstants.HTTP_MULTI_STATUS
+    private fun isSuccess(status: Int) = status == HttpConstants.HTTP_OK
```
Comment (abelgardep) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/GetUrlToOpenInWebRemoteOperation.kt:80]: 🥇

**Review by abelgardep — COMMENTED** [review]
Some suggestions here 🍻

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/CreateRemoteFileWithAppProviderOperation.kt (hunk 2 above)
Comment (JuancaG05) [reply owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/CreateRemoteFileWithAppProviderOperation.kt:47]: I used it like that as one single word 😁
https://www.wordreference.com/es/translation.asp?tranword=filename

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/CreateRemoteFileWithAppProviderOperation.kt (hunk 1 above)
Comment (JuancaG05) [reply owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/CreateRemoteFileWithAppProviderOperation.kt:106]: Yes, I think that's a good idea, but I see there are lots of timeout variables with different values, and some of them are defined as seconds and others as milliseconds (even sometimes I see 10_000 seconds, I think that's not correct). We would need to review this but I think it's out of the scope of this PR. Feel free to open a new issue with this if you want 🍻

**Review by jabarros — APPROVED** [review]
LGTM

**Pull request merged**